### PR TITLE
Add RFC process

### PR DESCRIPTION
**Pull request by am357**
*Issue #, if available:* #5 

*Description of changes:*
Adds the draft for RFC process for review.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: RFCs/0000-rfc-process.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+### Summary
+
+This RFC (request for comment) outlines the process that PartiQL maintainers and contributors should follow in order to introduce new changes to the language.
+
+This RFC process is inspired by [Rust RFCs.](https://github.com/rust-lang/rfcs)
+
+### Motivation
+The motivation is to provide governance to changes that require to be made to PartiQL so that the changes to PartiQL can be reviewed in a more consistent and controlled manner.
+
+*Change to PartiQL* is any change to PartiQL Specification, any of its implementations, this RFC, or changes affecting long term decision-making for maintaining and evolving PartiQL open source project.
```
Comment (RCHowell) [inline RFCs/0000-rfc-process.md:10]:
```suggestion
*Change to PartiQL* is any change to the PartiQL Specification, any of its implementations, this RFC, or changes affecting long term decision-making for maintaining and evolving the PartiQL open source project.
```

File: RFCs/0000-rfc-process.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+# Guide-level explanation
+The changes that require RFC are but not limited to the following:
```
Comment (RCHowell) [inline RFCs/0000-rfc-process.md:13]:
```suggestion
Changes that require RFC are, but not limited to, the following:
```

File: RFCs/0000-rfc-process.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+* Any semantic or syntactic changes to the language that is not a bugfix—e.g. addition of temporal table semantics.
+* Any changes to the public APIs—e.g. addition of new planner API.
+* Any backward incompatible changes—e.g. deprecating `ExprNode`.
```
Comment (RCHowell) [inline RFCs/0000-rfc-process.md:17]:
```suggestion
* Any semantic or syntactic changes to the language that are not bug fixes—e.g. addition of temporal table semantics.
* Any changes to the public APIs—e.g. addition of new planner API.
* Any backward incompatible changes—e.g. deprecating `ExprNode`.
```

File: RFCs/0000-rfc-process.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+3. Complete the RFC text using the template.
+4. Create a pull request to this repository.
+5. Add the pull request number to the RFC filename and pull request address to the placeholder in the template.
+6. Submit a new commit.
+7. The request enters the review process using the pull request.
+8. PR get closed upon the outcome of review is reached.
+
+### Approved proposals
+
+After the proposal is approved by a member of PartiQL committee:
```
Comment (RCHowell) [inline RFCs/0000-rfc-process.md:42]:
```suggestion
After the proposal is approved by a member of the PartiQL committee:
```

File: RFCs/0000-rfc-process.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+1. The PR gets merged with the pull request number as its Id.
```
Comment (RCHowell) [inline RFCs/0000-rfc-process.md:44]: nit, I don't think `Id` needs capitalization

File: RFCs/0000-rfc-process.md (hunk 2 above)
Comment (am357) [reply RFCs/0000-rfc-process.md:13]: I think we can neglect the commas here.

File: RFCs/0000-rfc-process.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+* Rephrasing, rewording the existing specification where the change does not affect the semantic—e.g. making ORDER BY section more clear.
+* Bug fixes.
+* Implementing feature that are already exist in PartiQL specification—e.g. implementing ORDER BY.
+* Performance improvements—e.g. improve the PartiQL’s CLI performance.
+* Changes to back-end that does not break consumers.
+
+## The Process
+
+Before submitting the RFC, try to discuss your idea using a pre-RFC document with PartiQL maintainers and community via [PartiQL Forum](https://community.partiql.org/faq)or [GitHub](https://github.com/partiql/).  This helps to solicit an early feedback and getting a sense of how the proposed change is going to be seen by PartiQL maintainers and community.
```
[the remark this answers is not in the document]
Comment (am357) [reply RFCs/0000-rfc-process.md:29]: Not the fan of double spaces :) will revise the other changes as per your suggestion.

File: RFCs/0000-rfc-process.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+1. Fork this repository at https://github.com/partiql/partiql-doc
+2. Copy `templates/0000-template.rst` to `RFCs/0000-RFC-NAME.rst`; do not assign a number yet.
```
[the remark this answers is not in the document]
Comment (am357) [reply RFCs/0000-rfc-process.md:32]: Yes, I agree with you. This is a miss from my side; should be `md`.

File: RFCs/0000-rfc-process.md (hunk 5 above)
Comment (am357) [reply RFCs/0000-rfc-process.md:44]: It's a name, so no?

File: RFCs/0000-rfc-process.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+* Any changes to the public APIs—e.g. addition of new planner API.
+* Any backward incompatible changes—e.g. deprecating `ExprNode`.
+
+The changes that don’t require RFC are but not limited to the following:
+
+* Rephrasing, rewording the existing specification where the change does not affect the semantic—e.g. making ORDER BY section more clear.
+* Bug fixes.
+* Implementing feature that are already exist in PartiQL specification—e.g. implementing ORDER BY.
+* Performance improvements—e.g. improve the PartiQL’s CLI performance.
+* Changes to back-end that does not break consumers.
```
[the remark this answers is not in the document]
Comment (am357) [reply RFCs/0000-rfc-process.md:25]: I think there are redundant commas in the first line, I'll incorporate your other suggestions here.

File: RFCs/0000-rfc-process.md (hunk 5 above)
Comment (RCHowell) [reply RFCs/0000-rfc-process.md:44]: I see. ID or `Id` is fine. No issue

**Review by RCHowell — APPROVED** [review]

**Pull request merged**